File: app/src/main/java/lemond/annoying/gamerscompanion/main_activity/fragment_news/viewmodel/NewsFragmentViewModelFactory.java
package lemond.annoying.gamerscompanion.main_activity.fragment_news.viewmodel;


import android.arch.lifecycle.ViewModel;
import android.arch.lifecycle.ViewModelProvider;
import android.support.annotation.NonNull;

import javax.inject.Inject;

import lemond.annoying.gamerscompanion.main_activity.fragment_news.injection.NewsFragmentScope;
import lemond.annoying.gamerscompanion.main_activity.fragment_news.model.NewsRepository;

@NewsFragmentScope
public class NewsFragmentViewModelFactory implements ViewModelProvider.Factory {

    private final NewsRepository newsRepository;

    @Inject
    NewsFragmentViewModelFactory(NewsRepository newsRepository) {
        this.newsRepository = newsRepository;
    }

    @Override
    @NonNull
    @SuppressWarnings("unchecked")
    public <T extends ViewModel> T create(@NonNull Class<T> modelClass) {
        if (modelClass.isAssignableFrom(NewsFragmentViewModel.class)) {
            return (T) new NewsFragmentViewModel(newsRepository);
        }
        throw new IllegalArgumentException("Unknown ViewModel class");
    }

}

File: app/src/main/java/lemond/annoying/gamerscompanion/main_activity/fragment_news/viewmodel/NewsFragmentViewModelFactory.kt
package lemond.annoying.gamerscompanion.main_activity.fragment_news.viewmodel


import android.arch.lifecycle.ViewModel
import android.arch.lifecycle.ViewModelProvider

import javax.inject.Inject

import lemond.annoying.gamerscompanion.main_activity.fragment_news.injection.NewsFragmentScope
import lemond.annoying.gamerscompanion.main_activity.fragment_news.model.NewsRepository

@NewsFragmentScope
class NewsFragmentViewModelFactory @Inject
internal constructor(private val newsRepository: NewsRepository) : ViewModelProvider.Factory {

    override fun <T : ViewModel> create(modelClass: Class<T>): T {
        if (modelClass.isAssignableFrom(NewsFragmentViewModel::class.java)) {
            return NewsFragmentViewModel(newsRepository) as T
        }
        throw IllegalArgumentException("Unknown ViewModel class")
    }

}
